Backend/Tournament_User/TournamentUserMVSC/server.js

```diff
@@ -0,0 +1,181 @@
+    
+    db.query(sql, [UUID, tournamentID, elo], (err, results) => {
+        if (err) {
+            // Handle MySQL errors, including the errors raised by SIGNAL in the stored procedure
+            if (err.sqlState === '45000') {
+                return res.status(400).send(err.message);  // Custom error from SIGNAL
+            }
+            console.error('Error during tournament signup:', err);
+            return res.status(500).send('Signup failed. Database error occurred');
+        }
+
+        // Respond with a success message
+        res.status(201).json({
+            message: 'Successfully signed up for the tournament',
+            details: { UUID, tournamentID, elo }
+        });
+    });
+});
+
+
+
+
+// Route to let user quit from a tournament
+app.delete('/tournaments/quit/:UUID', (req, res) => {
+    const UUID = req.params.UUID;   // UUID parsed from the URL
+    const { tournamentID } = req.body;  // tournamentID from the request body
+
+    // Check if required parameters are provided
+    if (!UUID || !tournamentID) {
+        return res.status(400).json({ error: 'Missing required fields (UUID or tournamentID)' });
+    }
+
+    // SQL query to delete the user from the Signups table
+    const deleteQuery = 'CALL UserQuit(?,?)';
+
+    // Execute the query to remove the user
+    db.query(deleteQuery, [UUID, tournamentID], (err, result) => {
+        if (err) {
+            console.error('Error deleting signup from the database:', err);
+            return res.status(500).send('Failed to remove the user from the tournament');
+        }
+
+        if (result.affectedRows === 0) {
+            return res.status(404).send('No signup found for the provided UUID and tournamentID');
+        }
+
+        // Respond with a confirmation message
+        res.status(200).json({
+            message: 'Successfully quit from the tournament'
+        });
+    });
+});
+
+
+app.get('/tournaments/upComing',(req ,res)=>{
+    const upcomingQuery = 'CALL GetUpComingTournament()'
+
+    db.query(upcomingQuery,(err,results)=>{
+        if(err){
+            console.error('Error retrieving upcoming tournament.');
+            return res.status(500).send('Database error occurred');
+        }
+
+        if(results.length===0){
+            return res.status(404).send('No upcoming tournament.')
+        }
+
+        // Respond with the list of upcoming tournaments
+        res.status(200).json({
+            message: 'Upcoming tournaments',
+            tournaments: results
+        });
+
+    });
+});
+
+app.get('/tournaments/inProgress',(req,res)=>{
+    const inProgressQuery = 'CALL GetInProgressTournament()'
+
+    db.query(inProgressQuery,(err,results)=>{
+        if(err){
+            console.error('Error retrieving inProgress tournament');
+            return res.status(500).send('Database Error occured.');
+        }
+
+        if(results.length===0){
+            return res.status(404).send('No inprogress tournament')
```

Comment:
i dont think this constitutes as an error so the `res.status` may have to change